### PR TITLE
[ALLUXIO-1733] Unifying client-side and server-side options idiom

### PR DESCRIPTION
**Pull request by jsimsa**
Addresses: https://tachyon.atlassian.net/browse/ALLUXIO-1733

### CODE REVIEW

**Comment by gpang** [comment]
LGTM

**Comment by jsimsa** [comment]
@calvinjia PTAL

**Comment by jsimsa** [comment]
@calvinjia implemented the change to Inode, InodeFile, and InodeDirectory we talked about offline, PTAL

File: core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java [hunk 1]
```diff
@@ -91,48 +47,36 @@ public Object getFieldValue(Inode o) {
...
    */
-  public synchronized void addChild(Inode child) {
-    mChildren.add(child);
+  public InodeDirectory(long id) {
+    super(id);
+    mDirectory = true;
+    mMountPoint = false;
+  }
+
+  private InodeDirectory(long id, long creationTimeMs) {
```
Comment (calvinjia) [inline core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java]: Why do we need this additional constructor?

Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java]: it comes down to whether we want creation time to be immutable ... similarly for id, if we do not need it to be immutable, then we can just use the default constructor and add setters for id and creation time

**Comment by calvinjia** [comment]
@jsimsa looks good overall, just have a question about creation time.

File: core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java (hunk 1 above)
Comment (calvinjia) [inline core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java]: I'm fine with having it as an immutable field.

**Comment by calvinjia** [comment]
LGTM

**Comment by gpang** [comment]
LGTM

**Pull request merged**